File: Python/MakeSumDivisiblebyp.py
class Solution(object):
    def minSubarray(self, nums, p):
        total = sum(nums)
        modulo = total % p
        if modulo == 0:
            return 0
        
        prefix_sum = 0
        min_length = len(nums)
        n = min_length
        prefix_dict = {0: -1} 
        
        for i, num in enumerate(nums):
            prefix_sum += num
            current_mod = prefix_sum % p
            target_mod = (current_mod - modulo) % p
            
            if target_mod in prefix_dict:
                min_length = min(min_length, i - prefix_dict[target_mod])
            
            prefix_dict[current_mod] = i
        
        return min_length if min_length < n else -1

File: C++/MakeSumDivisiblebyP.cpp
class Solution {
public:
    int minSubarray(vector<int>& nums, int p) {
        long long total = 0;
        for (int num : nums) total += num;

        int modulo = total % p;
        if (modulo == 0) return 0;

        unordered_map<int, int> prefix;
        prefix[0] = -1;

        long long prefix_sum = 0;
        int min_len = nums.size();
        for (int i = 0; i<nums.size(); i++) {
            prefix_sum += nums[i];
            int current_mod = prefix_sum % p;
            int target_mod = (current_mod - modulo + p) % p;

            if (prefix.count(target_mod)) min_len = min(min_len, i - prefix[target_mod]);

            prefix[current_mod] = i;
        }

        return min_len == nums.size() ? -1 : min_len;
    }
};
